external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala

```diff
@@ -538,6 +538,17 @@ private[kafka010] object KafkaSourceProvider extends Logging {
       .setIfUnset(ConsumerConfig.RECEIVE_BUFFER_CONFIG, 65536: java.lang.Integer)
       .build()
 
+  /**
+   * Returns a unique consumer group (group.id), allowing the user to set the prefix of
+   * the consumer group
+   */
+  private def streamingUniqueGroupId(parameters: Map[String, String],
+                                     metadataPath: String): String = {
+    val groupIdPrefix = parameters
+      .getOrElse("group.id.prefix", "spark-kafka-source")
```

Comment:
It seems like convention has been to mostly use camelcase for streaming options that aren't from the existing kafka.blah.whatever configuration namespace... e.g. subscribePattern, startingOffsets, maxOffsetsPerTrigger

Reply:
kafka.* is reserved for the existing kafka project's client configs, see e.g. line 86.  I'd just go with groupIdPrefix

Reply:
Makes sense, fixed in 39424dd